src/iop/watermark.c

```diff
@@ -381,7 +382,8 @@ static gchar *_watermark_get_svgdoc(dt_iop_module_t *self, dt_iop_watermark_data
 
   // EXIF datetime
   struct tm tt_exif = { 0 };
-  if(sscanf(image->exif_datetime_taken, "%d:%d:%d %d:%d:%d", &tt_exif.tm_year, &tt_exif.tm_mon,
+  dt_datetime_img_to_exif(image, datetime);
+  if(sscanf(datetime, "%d:%d:%d %d:%d:%d", &tt_exif.tm_year, &tt_exif.tm_mon,
```

Comment:
Maybe we can provide a simple API for this in datetime.h/c:

`dt_datetime_to_tm(char *datetime, struct tm *tt);`

or in one shot:

`dt_datetime_img_to_tm(const dt_image_t *, struct tm* tt);`

This will simplify some calls already and avoid error prone scanf() and I'm sure this will be reused more at some point.

Reply:
I agree on the principle but I'm not sure tm will be necessary.
I wanted to here to encapsulate img->datetime in a first time.
Then add msec to img->datetime and use it where useful (functions like variables.c and watermark won't use time_t/tm any more).
Last clean up the rest of the code. Probably file related datetime will continue to use time_t (but maybe not tm).
Can I add the necessary routines in a further PR ?

Reply:
> Can I add the necessary routines in a further PR ?

If we do not forget about this, yes! To me scanf() is so error prone and not easy to get right I really think it should be out of the view of devs and cleanly integrated into this new API.

Reply:
I see. Let me clean up that then.

Reply:
done